### PR TITLE
[FIX] hr_attendance: geolocate check in/out in kiosk mode

### PR DESCRIPTION
**Pull request by taqu-odoo**
Issue
-----
GPS coordinates are missing when an employee checks in/out in Kiosk mode.

Steps
-----
- Install Attendances
- Check in or check out from kiosk mode
- Go to Attendances > click on the corresponding attendance > the GPS coordinates are 0, 0 for any check in/out in "kiosk" mode.

opw-4239596

### CODE REVIEW

**Comment by taqu-odoo** [comment]
Not sure if this is wanted functionally, could be useful if multiple kiosk locations?

File: addons/hr_attendance/controllers/main.py [hunk 1]
```diff
@@ -138,11 +138,15 @@ def scan_barcode(self, token, barcode):
 
     @http.route('/hr_attendance/manual_selection', type="json", auth="public")
     def manual_selection(self, token, employee_id, pin_code):
+        return self.manual_selection_with_geolocation(token, employee_id, pin_code)
+
+    @http.route('/hr_attendance/manual_selection_with_geolocation', type="json", auth="public")
+    def manual_selection_with_geolocation(self, token, employee_id, pin_code, latitude=False, longitude=False):
```
Comment (nle-odoo) [inline addons/hr_attendance/controllers/main.py:144]: it's good to duplicate the method for stable I think (so people overriding the original ones), for the route I don't know if it's necessary or if we could just keep the route "/hr_attendance/manual_selection" on the new method

**Review by nle-odoo — APPROVED** [review]
seeing the rest of hr_attendance, it seems it could makes sense functionally to want this

as a side note: just looking at it, `self._get_geoip_response` should fallback to the latitude of request.geoip.location.latitude and request.geoip.location.longitude if you don't give it one. So as far as I can tell, if you [setup the Geo IP database](https://www.odoo.com/documentation/18.0/administration/on_premise/geo_ip.html#installation) it should not be 0 but the latitude/longitude of the location corresponding to your IP.

so I'm not sure if the issue happens on our Saas / Odoo.sh where I imagine the Geo IP is configured correctly (but if we use the localization from the browser in other case in hr_attendance, I guess it is preferred to Geo IP database).

**Review by Bertrand2 — APPROVED** [review]
LGTM, thanks :)
robodoo r+

**Pull request closed without merging**